### PR TITLE
chore: Use createContentDigest helper

### PR DESCRIPTION
**Pull request by sidharthachatterjee**
- [x] packages/gatsby-theme-blog-core/gatsby-node.js
- [x] packages/gatsby-source-contentful/src/normalize.js
- [x] packages/gatsby-plugin-mdx/gatsby/on-create-node.js
- [x] packages/gatsby-plugin-mdx/utils/create-fake-file-node.js

Fixes https://github.com/gatsbyjs/gatsby/issues/8805

Paired with @kindavishal on a community pairing session! 🎉

### CODE REVIEW

**Comment by sidharthachatterjee** [comment]
- [ ] Add `gatsby-core-utils` to dependencies where required

**Comment by muescha** [comment]
i remember the contentful was declined before #14567 but i think we should include it anyway to complete the issue...

**Comment by sidharthachatterjee** [comment]
Yeah, I don’t think we should block this on any PRs anymore. Let’s wrap it up.

File: packages/gatsby-plugin-mdx/utils/create-fake-file-node.js [hunk 1]
```diff
@@ -26,14 +25,8 @@ exports.createFileNode = async (
...
   if (stats.isDirectory()) {
-    const contentDigest = crypto
-      .createHash(`md5`)
-      .update(
-        JSON.stringify({ stats: stats, absolutePath: slashedFile.absolutePath })
-      )
-      .digest(`hex`)
     internal = {
-      contentDigest,
+      contentDigest : createContentDigest(obj),
```
Comment (sidharthachatterjee) [inline packages/gatsby-plugin-mdx/utils/create-fake-file-node.js]: Don't see an `obj` variable here. Shouldn't that be `{ stats: stats, absolutePath: slashedFile.absolutePath }`?

Comment (kindavishal) [reply packages/gatsby-plugin-mdx/utils/create-fake-file-node.js]: Thanks for pointing out, I will fix it.

Comment (kindavishal) [reply packages/gatsby-plugin-mdx/utils/create-fake-file-node.js]: https://github.com/gatsbyjs/gatsby/pull/20239/commits/8335edfeda7200070b6907a503b2e5ce2d02f0e8 fixes this

**Comment by kindavishal** [comment]
Hey @sidharthachatterjee can you tell me what I am doing wrong here?

File: packages/gatsby-source-contentful/src/normalize.js [hunk 2]
```diff
@@ -1,12 +1,9 @@
 const _ = require(`lodash`)
-const crypto = require(`crypto`)
 const stringify = require(`json-stringify-safe`)
+const deepMap = require(`deep-map`)
```
Comment (sidharthachatterjee) [inline packages/gatsby-source-contentful/src/normalize.js]:
```suggestion
const deepMap = require(`deep-map`)
```

Unsure where this came from. Linting says that it is unused. Should be good to remove.

Comment (kindavishal) [reply packages/gatsby-source-contentful/src/normalize.js]: https://github.com/gatsbyjs/gatsby/pull/20239/commits/a82d4c45216038fded6ddd11a7ec00b8034eeae1 fixes this

**Comment by kindavishal** [comment]
All checks should completed by now. Please let me know if I left anything or did something wrong somewhere.

**Comment by sidharthachatterjee** [comment]
@kindavishal Need to add dependencies in package.json. See previous comment.

**Comment by kindavishal** [comment]
Both have ```"peerDependencies": {
    "gatsby": "^2.12.1"
  }``` and ```"peerDependencies": {
    "gatsby": "^2.13.13",``` in their package.json files, is there something else required to add?

**Pull request merged**